internal/dot/graph.go

```diff
@@ -28,26 +50,34 @@ type Node struct {
 	Group    string
 }
 
-// Edge connects a node parameter to a node result.
-type Edge struct {
-	Param  *Node
-	Result *Node
+// Add updates the param and result nodes in the constructor and adds the constructor to the graph
+func (dg *Graph) Add(c *Ctor, params []*Node, results []*Node) {
+	c.Params = getNodes(dg, params)
+	c.Results = getNodes(dg, results)
+	dg.Ctors = append(dg.Ctors, c)
 }
 
```

Comment:
It might be useful to add a note on why you aren't re-using the `dig.node.String` method here (i.e. a note on the optional field).